From ae9b47daf95ea1808190726d32e93c92ab7563de Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Wed, 26 Feb 2014 10:27:08 +0000
Subject: [PATCH 137/351] switch: arizona: Clear detecting before we begin
 headphone detection

The detecting flag refers to the mic detection phase and as such should
be cleared before we begin the headphone detection phase.

Change-Id: Ic901ba54e02c8b297d1fe4d87f8440a8de4176fb
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index f04683b..91678e7 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -1018,15 +1018,15 @@ static void arizona_micd_detect(struct work_struct *work)
 					ret);
 			}
 		}
-		arizona_identify_headphone(info);
 		info->detecting = false;
+		arizona_identify_headphone(info);
 		goto handled;
 	}
 
 	/* If we got a high impedence we should have a headset, report it. */
 	if (info->detecting && (val & ARIZONA_MICD_LVL_8)) {
 		info->mic = true;
-
+		info->detecting = false;
 		arizona_identify_headphone(info);
 
 		/* Don't need to regulate for button detection */
@@ -1036,7 +1036,6 @@ static void arizona_micd_detect(struct work_struct *work)
 				ret);
 		}
 
-		info->detecting = false;
 		goto handled;
 	}
 
@@ -1049,10 +1048,11 @@ static void arizona_micd_detect(struct work_struct *work)
 	if (info->detecting && (val & MICD_LVL_1_TO_7)) {
 		if (info->jack_flips >= info->micd_num_modes * 10) {
 			dev_dbg(arizona->dev, "Detected HP/line\n");
-			arizona_identify_headphone(info);
 
 			info->detecting = false;
 
+			arizona_identify_headphone(info);
+
 			arizona_stop_mic(info);
 		} else {
 			info->micd_mode++;
-- 
1.7.9.5

